zhmcclient/_metrics.py

```diff
@@ -0,0 +1,390 @@
+# Copyright 2016 IBM Corp. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A Metrics Service provides a mechanism to retrieve performance
+metric data for resources of a physical z Systems or LinuxONE computer.
+A structure called a :term:`Metrics Context` is associated with
+any metrics retrieval, and that structure includes metric group names,
+individual metric field names, and the associated individual
+metric data types.
+"""
+
+from __future__ import absolute_import
+
+from ._manager import BaseManager
+from ._resource import BaseResource
+from ._logging import get_logger, logged_api_call
+from ._exceptions import NotFound
+
+__all__ = ['MetricsContextManager', 'MetricsContext', 'Metrics',
+           'CollectedMetrics']
+
+LOG = get_logger(__name__)
+
+
+class MetricsContextManager(BaseManager):
+    """
+    Manager providing access to the :term:`Metric sContexts <Metrics Context>`
+    exposed by the HMC this client is connected to.
+
+    Derived from :class:`~zhmcclient.BaseManager`; see there for common methods
+    and attributes.
+
+    Objects of this class are not directly created by the user; they are
+    accessible as properties in higher level resources (in this case,
+    the :class:`~zhmcclient.Client` object connecting to the HMC).
+    """
+
+    def __init__(self, client):
+        # This function should not go into the docs.
+        # Parameters:
+        #   client (:class:`~zhmcclient.Client`):
+        #      Client object for the HMC to be used.
+
+        # Resource properties that are supported as filter query parameters
+        # (for server-side filtering).
+        query_props = [
+        ]
+
+        super(MetricsContextManager, self).__init__(
+            resource_class=MetricsContext,
+            parent=None,
+            uri_prop='object-uri',
+            name_prop='',
+            query_props=query_props)
+
+        self._session = client.session
+        self._metrics_contexts = []
+        self._client = client
+
+    @logged_api_call
+    def list(self, full_properties=False):
+        """
+        List the MetricsContext exposed by the HMC this client is connected to.
+
```

Comment:
I first thought that having a `list()` function does not work because the HMC does not allow listing them, but at a second thought this seems to work.

However: We need to better point this out in the description of `list()`, e.g.:

    List the :term:`Metrics Context`s that have been created through this session.